editor/editor.tsx

```diff
@@ -169,19 +180,31 @@ const fetchSpecFromGist = async (gist: string) => {
     if (!dataFile) return Promise.reject(new Error('Gist does not contain a Gosling spec.'));
 
     const specUrl = new URL(`https://gist.githubusercontent.com/${gist}/raw/${dataFile}`);
-    const whenCode = fetch(specUrl.href).then(async response =>
-        response.status === 200 ? resolveRelativeCsvUrls(await response.text(), specUrl) : null
-    );
+    const whenCode = fetch(specUrl.href).then(async response => (response.status === 200 ? response.text() : null));
 
     const whenText = fetch(`https://gist.githubusercontent.com/${gist}/raw/${textFile}`).then(response =>
         response.status === 200 ? response.text() : null
     );
 
-    return Promise.all([whenCode, whenText]).then(([code, description]) => ({
-        code,
-        description,
-        title: metadata.description
-    }));
+    return Promise.all([whenCode, whenText]).then(([code, description]) => {
+        let jsonCode: string | null, jsCode: string, language: string;
+        if (isJSON(code)) {
+            language = 'json';
+            jsonCode = resolveRelativeCsvUrls(code, specUrl);
+            jsCode = json2js(jsonCode);
+        } else {
+            jsCode = code || '';
+            jsonCode = '{}';
+            language = 'javascript';
+        }
```

Comment:
Given that `code` can be either `null` or JS in this code block, it would be better to separate the two cases. 

If `code` is not found, we can set the `language` as 'json' since it is the default one of the Editor.

If `code` is JS, I think we need to set the JSON spec that is generated by the JS code, instead of the empty spec ('{}').